enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,340 @@
+Spec:
+  Node Ref:
+    API Version:  
+    Kind:         Node
+    Name:         ip-10-0-12-194.ec2.internal
+  Pool:
+    API Version:  machineconfiguration.openshift.io/v1
+    Kind:         MachineConfigPool
+    Name:         master
+Status:
+  Conditions:
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               Node has completed update to config rendered-master-cf99e619747ab19165f11e3546c71f1e
+    Reason:                NodeUpgradeComplete
+    Status:                True
+    Type:                  Updated
+    Last Transition Time:  2023-10-17T13:09:02Z
+    Message:               This node has not yet entered the UpdatePreparing phase
+    Reason:                NotYetOccured
+    Status:                False
+  Config Version:
+    Current:            rendered-worker-823ff8dc2b33bf444709ed7cd2b9855b
+    Desired:            rendered-worker-823ff8dc2b33bf444709ed7cd2b9855b
+  Health:               Healthy
+  Most Recent Error:    
+  Observed Generation:  3
+```
+
+The above struct gives us some helpful information about a node as it pertains to the MCO. First, all upgrade related events that have occured on the node from the most recent uprade process (no matter how small) appear in the `conditions`. Other than that, you can see the current and desired machineconfig indicating whether or not the node should be updating as well as whether or not the currently tracked update process held in `conditions` is updating to the expected MachineConfig. The Node refernece and ObservedGeneration exist to let the user know some extra information on the object and how many times we have gone throgh some upgrade related changes.
+
+The states to be reported by this MachineConfigNode will roughly fall into the following:
+
+#### Preparing phase
+ - stopping config drift monitor
+ - reconciling configs
+#### Actionable phase
+- Draining
+- Updating on disk state
+- Updating OS
+#### Post update action
+- Rebooting
+- Closing daemon
+- Node Reboot **OR** Reloading crio
+
+This loops until we are inDesiredConfig
+
+More Paths that will update the node state:
+
+#### Completing Update -- the daemon only gets into this phase (outside of hypershift) when we are inDesiredConfig,it has to do with the drainer annotation
+- Uncordoning
+- Uncordoned
+- Updating node state and metrics
+   		 
+#### Resume phase
+- Start config drift monitor
+- Error states:
+   - update stuck -- specifically degraded and/or failures in the functions this is the phase that happens if we either error in any of the above stages
+   - unavailable nodes... an obstacle has been hit, but its not the MCOs fault
+     - Disk Pressure
+   	 - Unschedulable
+
+
+### Workflow Description
+
+When an upgrade is triggered by there being mismatch between a desired and current config or simply just a new MachineConfig being applied, the MachineConfigNodes for a specific pool will report the following processes (roughly)
+
+
+```console
+$ oc get machineconfignodes
```

Comment:
I think when an upgrade gets triggered, all the nodes need to be updated to latest config. Hence, reporting `UPDATED` as `false` for all nodes would be more aligned. Once a node has been updated, they get set to `true`

Reply:
Alternatively, maybe instead of `updating`, what we really want is `idle`? Then the user can still gauge updated status via MCP and use this to see what nodes are in action

Reply:
I'm open to Idle, that seems like it could be helpful. What would trigger Idle to go into false? 

I had expected, kind of as @sinnykumari mentions

```
Name.           Updated.   Updating
my-node-1.   False.         True
my-node-2.  False.          False
````

Reply:
I can make a card to track an idle state, though that will cause unnecessary API calls that might have a "blip" effect where Updated only is true for a fraction of a second. Might instead want to switch updated to idle